### PR TITLE
win_timezone: Clean up and check-mode support

### PR DESCRIPTION
**Pull request by dagwieers**
##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

win_timezone

##### ANSIBLE VERSION

v2.3

##### SUMMARY

Changes include:
- Use Get-AnsibleParam with -type support
- Replace $result PSObject with normal hash
- Add check-mode support

### CODE REVIEW

**Comment by dagwieers** [comment]
ready_for_review

**Comment by schwartzmx** [comment]
lgtm

shipit

File: lib/ansible/modules/windows/win_timezone.ps1 [hunk 1]
```diff
@@ -19,28 +19,32 @@
...
+$result = @{
     changed = $false
+    win_timezone = @{}
 }
 
-$timezone = Get-Attr -obj $params -name timezone -failifempty $true -resultobj $result
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$timezone = Get-AnsibleParam -obj $params -name "timezone" -failifempty $true -resultobj $result
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_timezone.ps1:30]: Do we want to change this and remove the -resultobj to keep in line with the rest of the modules?

File: lib/ansible/modules/windows/win_timezone.ps1 [hunk 2]
```diff
@@ -19,28 +19,32 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
-$params = Parse-Args $args;
-
-$result = New-Object psobject @{
-    win_timezone = New-Object psobject
+$result = @{
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_timezone.ps1:22]: $ErrorActionPreference = "Stop" like other modules?

File: lib/ansible/modules/windows/win_timezone.ps1 [hunk 3]
```diff
@@ -19,28 +19,32 @@
...
 Try {
     # Get the current timezone set
     $currentTZ = $(tzutil.exe /g)
-    If ($LASTEXITCODE -ne 0) { Throw "An error occurred when getting the current machine's timezone setting." /
+    If ($LASTEXITCODE -ne 0) {
+        Throw "An error occurred when getting the current machine's timezone setting."
+    }
 
     If ( $currentTZ -eq $timezone ) {
         Exit-Json $result "$timezone is already set on this machine"
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_timezone.ps1:40]: Do we need this here or should we just remove this if branch and just exit at the end of the code like normal

File: lib/ansible/modules/windows/win_timezone.ps1 [hunk 4]
```diff
@@ -49,23 +53,28 @@ Try {
...
-        }
-        Else {
+        } Else {
             Fail-Json $result "The specified timezone: $timezone isn't supported on the machine."
         }
     }
-}
-Catch {
+} Catch {
     Fail-Json $result "Error setting timezone to: $timezone."
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_timezone.ps1:77]: Add in $($_.Error.Message) to the end for better error handling.

**Review by jborean93 — CHANGES_REQUESTED** [review]

**Comment by jborean93** [comment]
Hey @dagwieers just requested some changes to keep this more in line with the other work you have done, let me know your thoughts.

File: lib/ansible/modules/windows/win_timezone.ps1 (hunk 2 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_timezone.ps1:22]: Well, I didn't want to change the module in other ways then just the things I now changed.
Things like this we have to discuss with the Windows working group.

File: lib/ansible/modules/windows/win_timezone.ps1 (hunk 1 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_timezone.ps1:30]: To be honest, the -resultobj does not seem to do anything except be used for error-handling within the AnsibleParam function. So I would remove them myself, but I don't think I am smarter than the person who added it, so I restricted myself to the stuff I do know and are safe to change.

**Comment by dagwieers** [comment]
@jborean93 In general I wanted to restrict myself to AnsibleParam, $result hash and maybe $check_mode. Not a complete rewrite. In some cases I did modify the indentation/syntax, but mostly when that wasn't consistent within the file. I know it is frowned upon to change cosmetics.

**Review by jborean93 — APPROVED** [review]

**Comment by jborean93** [comment]
@dagwieers no worries I was unsure what the scope of your changes were, looks good.

shipit

**Comment by jhawkesworth** [comment]
Looks good to me and has 2 shipits, 1 from the maintainer, so I'm merging this in.

**Pull request merged**